server/src/com/thoughtworks/go/server/service/EnvironmentConfigService.java

```diff
@@ -145,6 +136,24 @@ public EnvironmentConfig named(String environmentName) throws NoSuchEnvironmentE
         return environments.named(new CaseInsensitiveString(environmentName));
     }
 
+    public EnvironmentConfig getEnvironmentConfig(String environmentName) throws NoSuchEnvironmentException {
+        if(environments.hasEnvironmentNamed(new CaseInsensitiveString(environmentName)))
+            return environments.named(new CaseInsensitiveString(environmentName));
+        return null;
+    }
+
+    public void deleteEnvironment(String environmentName, HttpLocalizedOperationResult result) throws NoSuchEnvironmentException {
+        CruiseConfig config = goConfigService.getConfigForEditing();
+        EnvironmentsConfig environments = config.getEnvironments();
+        if (environments.remove(getEnvironmentConfig(environmentName))) {
+            config.setEnvironments(environments);
+            this.environments.remove(getEnvironmentConfig(environmentName));
+            result.setMessage(LocalizedMessage.string("ENVIRONMENT_DELETE_SUCCESSFUL", environmentName));
+        } else {
+            result.notFound(LocalizedMessage.string("ENVIRONMENT_NOT_FOUND", environmentName), HealthStateType.general(HealthStateScope.GLOBAL));
```

Comment:
This feels misleading/confusing given what you're doing here https://github.com/gocd/gocd/pull/2207/files#diff-c13835adf2f105a5355767fdee02e76eR61